mithril-common/src/store/adapter/sqlite_adapter.rs

```diff
@@ -0,0 +1,500 @@
+use async_trait::async_trait;
+use serde::{de::DeserializeOwned, Serialize};
+use sha2::{Digest, Sha256};
+use sqlite::{Connection, State, Statement};
+use tokio::sync::{Mutex, MutexGuard};
+
+use std::{marker::PhantomData, path::PathBuf, sync::Arc};
+
+use super::{AdapterError, StoreAdapter};
+
+type Result<T> = std::result::Result<T, AdapterError>;
+
+/// Store adapter for SQLite3
+struct SQLiteAdapter<K, V> {
+    connection: Arc<Mutex<Connection>>,
+    table: String,
+    key: PhantomData<K>,
+    value: PhantomData<V>,
+}
+
+impl<K, V> SQLiteAdapter<K, V>
+where
+    K: Serialize,
+    V: DeserializeOwned,
+{
+    /// Create a new SQLiteAdapter instance.
+    pub fn new(table_name: &str, file: Option<PathBuf>) -> Result<Self> {
+        let file = file
+            .map(|v| v.to_string_lossy().to_string())
+            .unwrap_or_else(|| ":memory:".to_string());
+        let connection =
+            Connection::open(file).map_err(|e| AdapterError::InitializationError(e.into()))?;
```

Comment:
use a match here so you won't have to convert the PathBuf to a string (avoiding potential errors if the path contains some characters that don't translate well to a rust string) :
```rust
let connection = match file {
  Some(file) => Connection::open(file),
  None => Connection::open(":memory"),
}.map_err(|e| AdapterError::InitializationError(e.into()))?;
```